lib.rs

```diff
@@ -5,56 +5,85 @@
 //! They do not support deallocation of individual objects while the arena itself is still alive.
 //! The benefit of an arena is very fast allocation; just a vector push.
 //!
-//! This is an equivalent to [`arena::TypedArena`](http://doc.rust-lang.org/arena/struct.TypedArena.html)
+//! This is an equivalent of
+//! [`arena::TypedArena`](http://doc.rust-lang.org/arena/struct.TypedArena.html)
 //! distributed with rustc, but is available of Rust beta/stable.
 //!
-//! It is probably slightly less efficient, but is simpler internally and uses much less unsafe code.
+//! It is slightly less efficient, but simpler internally and uses much less unsafe code.
 //! It is based on a `Vec<Vec<T>>` instead of raw pointers and manual drops.
 
+// Potential optimizations:
+// 1) add and stabilize a method for in-place reallocation of vecs.
+// 2) add and stabilize placement new.
+// 3) use an iterator. This may add far too much unsafe code.
```

Comment:
I don’t know if this is desirable. If an iterator is added, `alloc` would have to be changed to return `&T` instead of `&mut T` to avoid introducing mutable aliasing. What do you think?